src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -969,6 +976,8 @@ private void CreateResponseHeader(bool appCompleted)
                 responseHeaders.SetRawDate(dateHeaderValues.String, dateHeaderValues.Bytes);
             }
 
+            _keepAlive = keepAlive;
```

Comment:
I'd rather leave it at the end, would be easy to overlook and make a change that sets the local again.

My only concern here is that something throws (could it happen?) and things are left in an inconsistent state.